zavod/zavod/tests/runtime/test_timestamps.py

```diff
@@ -26,3 +29,35 @@ def test_timestamps(vdataset: Dataset):
         assert index.get(stmt.id, default) == prev_time
 
     assert "TimeStampIndex" in repr(index), repr(index)
+
+
+def test_backfill(vdataset: Dataset):   
+    dataset_path = settings.DATA_PATH / "datasets" / vdataset.name
+    rmtree(dataset_path, ignore_errors=True)
+    
+    settings.ARCHIVE_BACKEND = "FilesystemBackend"
+    settings.ARCHIVE_PATH = Path(mkdtemp())
+    prev_time = settings.RUN_TIME_ISO
+    run_dataset(vdataset)
+
+    archive_path = settings.ARCHIVE_PATH / "datasets/latest/testdataset1"
+    archive_path.mkdir(parents=True, exist_ok=True)
+    copyfile(
+        settings.DATA_PATH / "datasets" / vdataset.name / "statements.pack",
+        archive_path / "statements.pack",
+    )
+
+    settings.RUN_TIME = settings.RUN_TIME + timedelta(days=1)
+    settings.RUN_TIME_ISO = settings.RUN_TIME.isoformat(sep="T", timespec="seconds")
+    settings.RUN_DATE = settings.RUN_TIME.date().isoformat()
+    second_time = settings.RUN_TIME_ISO
+    run_dataset(vdataset)
```

Comment:
by dong this, we're

1. creating a context which creates a backfilled index (context, timestamps.build)
2. emitting new statements (nomenklatura entity merge)

Reply:
does this stuff make sense when we're testing this at the TimestampIndex level? if we're not asserting that last_seen is different, there isn't really any point in running the dataset a second time, is there?